app/Actions/GenerateEventReport.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Actions;
+
+use App\Models\Character;
+use App\Models\Event;
+use App\Models\Session;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Str;
+use Lorisleiva\Actions\Concerns\AsAction;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class GenerateEventReport
+{
+    use AsAction;
+
+    /**
+     * Generates a CSV Report that tallies a users ratings on each dimension of their play style
+     *
+     * @param bool $onlyEventRatings
+     * @return StreamedResponse
+     */
+    public function handle(Event $event)
+    {
+        //prep
+        $columns = $this->prepareColumns();
+        $suffix = "event-report";
+
+        //get all characters that attended the event into one list
+        $characters = Character::whereHas(
+            'sessions',
+            function ($q) use ($event) {
+            $q->where('event_id', $event->id);
+        }
+        )->get();
```

Comment:
I think the PSR standard for call backs like this is the following format, or atleast thats the one i've always followed

```suggestion
        $characters = Character::whereHas('sessions', function ($q) use ($event) {
            $q->where('event_id', $event->id);
        })->get();
```

Reply:
weird, thats how I had it. Idk when it got changed but I definitely didnt format it that way.